query/Matcher.java

```diff
@@ -118,71 +109,13 @@ public static Matcher.Group.Aggregator create(Reasoner reasoner, TypeQLMatch.Gro
         return new Group.Aggregator(group, query);
     }
 
-    public FunctionalIterator<ConceptMap> execute() {
+    public FunctionalIterator<? extends ConceptMap> execute() {
         assert context != null;
         return execute(context);
     }
 
-    FunctionalIterator<ConceptMap> execute(Context.Query context) {
-        FunctionalIterator<ConceptMap> answers = reasoner.execute(disjunction, query.modifiers(), context);
-        // TODO: we should remove these and handle them in the traversal engine or reasoner ONLY. Currently in reasoner already
-        if (query.modifiers().sort().isPresent()) answers = sort(answers, query.modifiers().sort().get());
-        if (query.modifiers().offset().isPresent()) answers = answers.offset(query.modifiers().offset().get());
-        if (query.modifiers().limit().isPresent()) answers = answers.limit(query.modifiers().limit().get());
```

Comment:
sort, offset, and limit are all moved into the `Reasoner`. The sorting `Comparator` is moved into `ConceptMap.Ordered`